### ISSUE TITLE
Time consuming, so long

### ISSUE THREAD

**Issue by marson666** [ChenWu98/cycle-diffusion#8]
Hi, Chen, thanks for you great work.
I have tried to run zero-shot image-to-image translation with Stable Diffusion v1-4 with A100 single GPU, it costs 14.5 hours to finish the task, does this expected? Maybe the time consuming is too long.

`export CUDA_VISIBLE_DEVICES=0`
`export RUN_NAME=translate_text2img256_stable_diffusion_stochastic_1`
`export SEED=42`
`nohup python -m torch.distributed.launch --nproc_per_node 1 --master_port 1405 main.py --seed $SEED --cfg experiments/$RUN_NAME.cfg --run_name $RUN_NAME$SEED --logging_strategy steps --logging_first_step true --logging_steps 4 --evaluation_strategy steps --eval_steps 50 --metric_for_best_model CLIPEnergy --greater_is_better false --save_strategy steps --save_steps 50 --save_total_limit 1 --load_best_model_at_end --gradient_accumulation_steps 4 --num_train_epochs 0 --adafactor false --learning_rate 1e-3 --do_eval --output_dir output/$RUN_NAME$SEED --overwrite_output_dir --per_device_train_batch_size 1 --per_device_eval_batch_size 4 --eval_accumulation_steps 4 --ddp_find_unused_parameters true --verbose true > $RUN_NAME$SEED.log 2>&1 &`

**Comment by ChenWu98** [comment]
Hi,

For this experimental version, this is expected. The reason is: as we mentioned in the Appendix, for each sample, we enumerated a large number of hyperparameters and random seeds (this is also done for the baselines) to simulate the per-sample hyperparameter tuning and random trials in practice. 

I have implemented a version based on the [diffusers](https://github.com/huggingface/diffusers) library, called the CycleDiffusion pipeline. The code is already merged into their main branch. There, you can tune the hyperparameter and random seed yourself, and it only takes around **30 seconds** for each sample. In several days, I will create a Colab notebook and a Gradio demo based on my implementation in the [diffusers](https://github.com/huggingface/diffusers) library. Hope this is helpful.

**Comment by ChenWu98** [comment]
Just a follow-up here. We have updated the [**Diffusers 🧨 implementation**](https://huggingface.co/docs/diffusers/main/en/api/pipelines/cycle_diffusion) and the [**HuggingFace 🤗 demo**](https://huggingface.co/spaces/ChenWu98/Stable-CycleDiffusion).
